source/server/options_impl.cc

```diff
@@ -209,4 +216,41 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv,
     throw NoServingException();
   }
 }
+
+void OptionsImpl::parseComponentLogLevels(const std::string& component_log_levels) {
+  if (component_log_levels.empty()) {
+    return;
+  }
+  std::vector<std::string> log_levels = absl::StrSplit(component_log_levels, ',');
+  for (auto& level : log_levels) {
+    std::vector<std::string> log_name_level = absl::StrSplit(level, ':');
+    if (log_name_level.size() != 2) {
+      logError(fmt::format("error: component log level not correctly specified '{}'", level));
+    }
+    std::string log_name = log_name_level[0];
+    std::string log_level = log_name_level[1];
+    size_t level_to_use = std::numeric_limits<size_t>::max();
+    for (size_t i = 0; i < ARRAY_SIZE(spdlog::level::level_names); i++) {
```

Comment:
Do we already do this in admin for the `/logging` endpoint? Would a shared helper be useful here?

Reply:
Since the format is little different, shared helper may not be of much help here.